src/libraries/Common/src/System/Security/Cryptography/DSAAndroid.cs

```diff
@@ -0,0 +1,423 @@
+                }
+            }
+
+            private SafeDsaHandle GenerateKey()
+            {
+                SafeDsaHandle key;
+
+                if (!Interop.AndroidCrypto.DsaGenerateKey(out key, KeySize))
+                {
+                    throw new CryptographicException();
+                }
+
+                return key;
+            }
+
+            protected override byte[] HashData(byte[] data, int offset, int count, HashAlgorithmName hashAlgorithm)
+            {
+                // we're sealed and the base should have checked this already
+                Debug.Assert(data != null);
+                Debug.Assert(offset >= 0 && offset <= data.Length);
+                Debug.Assert(count >= 0 && count <= data.Length);
+                Debug.Assert(!string.IsNullOrEmpty(hashAlgorithm.Name));
+
+                return AsymmetricAlgorithmHelpers.HashData(data, offset, count, hashAlgorithm);
+            }
+
+            protected override byte[] HashData(Stream data, HashAlgorithmName hashAlgorithm) =>
+                AsymmetricAlgorithmHelpers.HashData(data, hashAlgorithm);
+
+            protected override bool TryHashData(ReadOnlySpan<byte> data, Span<byte> destination, HashAlgorithmName hashAlgorithm, out int bytesWritten) =>
+                AsymmetricAlgorithmHelpers.TryHashData(data, destination, hashAlgorithm, out bytesWritten);
+
+            public override byte[] CreateSignature(byte[] rgbHash)
+            {
+                if (rgbHash == null)
+                    throw new ArgumentNullException(nameof(rgbHash));
+
+                SafeDsaHandle key = GetKey();
+                int signatureSize = Interop.AndroidCrypto.DsaEncodedSignatureSize(key);
+                int signatureFieldSize = Interop.AndroidCrypto.DsaSignatureFieldSize(key) * BitsPerByte;
+                Span<byte> signDestination = stackalloc byte[SignatureStackBufSize];
+
+                ReadOnlySpan<byte> derSignature = SignHash(rgbHash, signDestination, signatureSize, key);
+                return AsymmetricAlgorithmHelpers.ConvertDerToIeee1363(derSignature, signatureFieldSize);
+            }
+
+#if INTERNAL_ASYMMETRIC_IMPLEMENTATIONS
+            public override bool TryCreateSignature(
+                ReadOnlySpan<byte> hash,
+                Span<byte> destination,
+                out int bytesWritten)
+            {
+                return TryCreateSignatureCore(
+                    hash,
+                    destination,
+                    DSASignatureFormat.IeeeP1363FixedFieldConcatenation,
+                    out bytesWritten);
+            }
+
+            protected override bool TryCreateSignatureCore(
+                ReadOnlySpan<byte> hash,
+                Span<byte> destination,
+                DSASignatureFormat signatureFormat,
+                out int bytesWritten)
+#else
+            public override bool TryCreateSignature(ReadOnlySpan<byte> hash, Span<byte> destination, out int bytesWritten)
+#endif
```

Comment:
The \#else case here never applies... so we should remove the conditional compilation